Dockerfile

```diff
@@ -2,11 +2,12 @@ FROM python:3.6
 MAINTAINER Zach White <skullydazed@gmail.com>
 
 WORKDIR /
-RUN git clone https://github.com/qmk/qmk_compiler_api.git
-WORKDIR /qmk_compiler_api
-RUN pip3 install git+git://github.com/qmk/qmk_compiler_worker.git@master
-RUN pip3 install git+git://github.com/skullydazed/kle2xy.git@master
+#RUN git clone https://github.com/qmk/qmk_api.git
+WORKDIR /qmk_api
+COPY . /qmk_api
 RUN pip3 install -r requirements.txt
+RUN pip3 install git+git://github.com/qmk/qmk_compiler.git@master
+RUN pip3 install git+git://github.com/skullydazed/kle2xy.git@master
```

Comment:
I'd move these from here to `requirements.txt` (just drop the initial `RUN pip3 install ` part).

Reply:
These don't have packages in pip yet. I know you can put URLs into requirements.txt but I prefer to keep them separate so they're visible as an "other" that needs to be folded in.

Reply:
If they're outside of `requirements.txt`, qmk_api won't work if started locally. They _are_ requirements, so why not have them as such?

There's also the option of installing them in editable mode (the equivalent of setuptools' [dependency links](https://python-packaging.readthedocs.io/en/latest/dependencies.html#packages-not-on-pypi)), which emphasizes that they're non-PyPI “other” projects, as you described.
```
-e git+git://github.com/qmk/qmk_compiler.git@master#egg=qmk_compiler
-e git+git://github.com/skullydazed/kle2xy.git@master#egg=kle2xy
```
I don't recommend this, however, as the `volumes` mount will override it, so it won't work.

Reply:
If you're not using docker you should be using the virtualenv setup script, which does handle that dependency.

I appreciate that you want to help, but you're missing a ton of context here.

Reply:
Is the `--editable` install crucial for local development (virtualenv w/o Docker)?

Reply:
The context is that there are 50,000 things I haven't had time to get to and this hasn't bubbled up to the top of the list. Changing this means possibly introducing bugs for reasons that have nothing to do with realizing the goals of this project.

Long term the changes you suggest will absolutely be made. Short term I'm more worried about getting things working. If you want to help, please direct this energy into [implementing features or fixing bugs](https://github.com/qmk/qmk_api/issues). There is a lot of heavy lifting to be done. Standing on the sidelines telling the people doing the work how to do that work is very frustrating for the people actually doing the work.

Reply:
I understand now. I've been trying to familiarize myself with how everything is set up, and the discussion around this PR proved to be a good starting point. I've been pointing out issues as I came across them. _Please_ don't take this as a sign of ill intent. I'm doing it because I plan on contributing more substantially once time allows. Otherwise, what would be the point?

If you'd like, I can make a PR for all the infrastructural changes I proposed here once we're at a point where you feel comfortable with the state of the project (when enough features have been implemented and bugs fixed), and then you can treat my code with the same amount of rigor I'm treating yours.

I'm deeply sorry if my remarks came across as rude and unappreciative.

Reply:
I appreciate that your intentions were good, and I'm hoping you'll still join us in working on this. It just seemed like you wanted to fix everything at once, but we're not really ready for changes on that scale yet. :)

The next step for cleaning up this particular issue is getting [/skullydazed/kle2xy](https://github.com/skullydazed/kle2xy) into pypi. I haven't touched that in a while and it looks like one of the tests needs to be fixed and probably some general cleanup could happen. Once that is done we can move this into requirements.txt.

Beyond that this discussion should probably move to a new issue or to discord.

Thanks for your understanding and apology.

Reply:
Agreed. Thank you for _your_ understanding.